build.gradle

```diff
@@ -361,8 +365,10 @@ subprojects {
         sign configurations.archives
     }
 
+    def publishGate = project.tasks.getByPath(':buildAndTestAll')
+    publishGate.dependsOn build, check
```

Comment:
huh, that has not been my experience in the past. I wonder if they changed it?
Ahh, I know what it is. Publish doesn't depend on build, it depends on assemble.